dotnet/src/Connectors/Connectors.OpenAI/Core/ClientCore.cs

```diff
@@ -88,7 +88,7 @@ internal ClientCore(
         HttpClient? httpClient = null,
         ILogger? logger = null)
     {
-        this.FunctionCallsProcessor = new FunctionCallsProcessor(this.Logger);
+        this.FunctionCallsProcessor = new FunctionCallsProcessor(logger);
```

Comment:
Recommend use the treated null logger before passing it to the `CallProcessor`.

```suggestion
        this.Logger = logger ?? NullLogger.Instance;
        this.FunctionCallsProcessor = new FunctionCallsProcessor(this.Logger);
```

Reply:
fixed